hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -393,6 +397,32 @@ public boolean delete(final Path f, final boolean recursive) throws IOException
 
   }
 
+  /**
+   * Returns a ContentSummary instance containing the count of directories,
+   * files and total number of bytes under a given path
+   * @param path The given path
+   * @return ContentSummary
+   * @throws IOException if an error is encountered during listStatus calls
+   * or if there is any issue with the thread pool used while processing
+   */
+  @Override
+  public ContentSummary getContentSummary(Path path) throws IOException {
+    try {
+      ABFSContentSummary contentSummary =
+          (new ContentSummaryProcessor(abfsStore)).getContentSummary(path);
```

Comment:
Can we return directly ContentSummary object, why we need to create ABFSContentSummary class ?

Reply:
ContentSummary class has extra members for storing and calculations related to snapshots of the directory structure, which are not needed here. Hence created a new class for ABFS